src/ui/RelevanceInspector/RelevanceInspector.ts

```diff
@@ -0,0 +1,171 @@
+    this.opened = true;
+
+    const content = $$('div');
+    const animation = DomUtils.getBasicLoadingAnimation();
+    content.append(animation);
+    this.modalBox.open(content.el, this.modalBoxOptions);
+    const rows = await this.buildTabs();
+    if (rows) {
+      animation.remove();
+      content.append(rows.el);
+    } else {
+      this.modalBox.close();
+    }
+  }
+
+  private handleNewResultDisplayed(args: IDisplayedNewResultEventArgs) {
+    if (this.bindings.queryStateModel.get(QueryStateModel.attributesEnum.debug)) {
+      $$(args.item).addClass('coveo-with-inline-ranking-info');
+      if ($$(args.item).hasClass('coveo-table-layout')) {
+        $$(args.item).append(new InlineRankingInfo(args.result).build().el);
+      } else {
+        $$(args.item).prepend(new InlineRankingInfo(args.result).build().el);
+      }
+    }
+  }
+
+  private get modalBoxOptions() {
+    return {
+      title: 'Relevance Inspector',
+      titleClose: false,
+      overlayClose: true,
+      sizeMod: 'big',
+      className: 'relevance-inspector-modal',
+      validation: () => {
+        this.opened = false;
+        return true;
+      }
+    };
+  }
+
+  private toggleFromState(stateValue: number) {
+    stateValue != 0 ? this.show() : this.hide();
+    this.modalBox.close();
+  }
+
+  private async buildTabs() {
+    const lastResults = this.bindings.queryController.getLastResults().results;
+    if (!lastResults || lastResults.length == 0) {
+      new Logger(this).error(
+        'Could not open Relevance Inspector because there is no results to display. Please execute a query beforehand',
+        lastResults
+      );
+      return;
+    }
+    if (!lastResults[0].rankingInfo) {
+      new Logger(this).error(
+        'Could not open Relevance Inspector because there is no ranking info returned on the results. Please execute a query in debug mode beforehad',
+        lastResults
+      );
+      return;
+    }
+
+    const container = $$('div');
+
+    const rankingInfoTable = await new RankingInfoTable(lastResults, this.bindings);
+    const metadataTable = await new MetaDataTable(lastResults, this.bindings);
+    const executionReport = await new ExecutionReport(this.bindings.queryController.getLastResults(), this.bindings);
+    const availableFields = await new AvailableFieldsTable(this.bindings);
+
+    this.tabs = {
+      relevanceInspectorRankingInfo: rankingInfoTable,
+      relevanceInspectorMetadata: metadataTable,
+      relevanceInspectorExecutionReport: executionReport,
+      relevanceInspectorAvailableFields: availableFields
+    };
+
+    const inspectorTabs = new RelevanceInspectorTabs(tabChangedTo => {
+      this.activeTab = tabChangedTo;
+      this.resize();
+    });
+
+    const debouncedResize = debounce(() => this.resize(), 100);
+
+    window.addEventListener('resize', debouncedResize);
+
+    inspectorTabs.addSection('Ranking Information', await rankingInfoTable.build(), 'relevanceInspectorRankingInfo');
```

Comment:
Could we launch these tasks in parallel with `Promise.all`? Would it be worth it?

Reply:
Hum we could, but I think the code would be less clear ?

 As in I'd have to store all promises, then get each promise content when they have all resolved to be able to append them correctly/with the correct title/in the correct section.

Either way, each request does not take a long time, it's just that TableBuilder (the underlying module that loads aggrig) needs to load the library at least once, so everything becomes async.